tools/massive_change.py

```diff
@@ -0,0 +1,221 @@
+                'public.')
+        self.gh_credentials = {'name': gh_user.name or str(gh_user),
+                               'email': gh_user.email or email}
+        self.gh_target_branch = target
+        self.gh_org = target_org or 'OCA'
+
+    def _replace_content(self, repo, path, replace_list, gh_file=None):
+        if not gh_file:
+            # Re-read path for retrieving content
+            gh_file = repo.file_contents(path, self.gh_target_branch)
+        content = gh_file.decoded.decode('utf-8')
+        for replace in replace_list:
+            new_content = re.sub(
+                replace[0], replace[1], content, flags=re.DOTALL)
+            if new_content == content:
+                return {}
+        new_file_blob = repo.create_blob(new_content, encoding='utf-8')
+        return {
+            'path': path,
+            'mode': '100644',
+            'type': 'blob',
+            'sha': new_file_blob,
+        }
+
+    def _create_commit(self, repo, tree_data, message, use_sha=True):
+        """Create a GitHub commit.
+        :param repo: github3 repo reference
+        :param tree_data: list with dictionary for the entries of the commit
+        :param message: message to use in the commit
+        :param use_sha: if False, the tree_data structure will be considered
+        the full one, deleting the rest of the entries not listed in this one.
+        """
+        if not tree_data:
+            return
+        if use_sha:
+            branch = repo.branch(self.gh_target_branch)
+            tree_sha = branch.commit.commit.tree.sha
+            parents = [branch.commit.sha]
+        else:
+            tree_sha = None
+            parents = []
+        tree = repo.create_tree(tree_data, tree_sha)
+        commit = repo.create_commit(
+            message=message, tree=tree.sha, parents=parents,
+            author=self.gh_credentials, committer=self.gh_credentials,
+        )
+        if use_sha:
+            repo.ref('heads/{}'.format(branch.name)).update(commit.sha)
+        return commit
+
+    def _do_change(self, project, path, src, dest, message=None):
+        print("Performing change in project %s/%s" % (self.gh_org, project))
+        repo = self.github.repository(self.gh_org, project)
+        try:
+            repo.branch(self.gh_target_branch)
+        except NotFoundError:
+            print("Branch doesn't exist. Skipping...")
+            return
+        root_contents = repo.directory_contents(
+            '', self.gh_target_branch, return_as=dict,
+        )
+        if not root_contents.get(path):
+            print("Path not found. Skipping...")
+            return
+        new_content = self._replace_content(repo, path, [(src, dest)])
+        if not new_content:
+            print("Source string not found. Skipping...")
+            return
+        tree_data = [new_content]
+        if not message:
+            message = DEFAULT_MESSAGE
+        self._create_commit(repo, tree_data, message)
```

Comment:
Could the option to, instead of committing directly, create a PR also be added here? https://github3py.readthedocs.io/en/master/api-reference/repos.html#github3.repos.repo.Repository.pull_request

Reply:
Not, this is not intended for creating PRs. There's no sense on creating 200 PRs, one per repo, for an administrative commit that fixes/updates anything broken. It has to be direct.